Code/autopkglib/URLDownloaderPython.py

```diff
@@ -0,0 +1,418 @@
+#!/usr/local/autopkg/python
+#
+# Copyright 2021 James Stewart @JGStew
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""See docstring for URLDownloaderPython class"""
+
+import json
+import os
+import ssl
+
+from hashlib import sha1, sha256, md5
+
+try:
+    from urllib.request import urlopen  # Python 3
+except ImportError:
+    print("ERROR: Not tested with Python 2")
+    from urllib2 import urlopen  # Python 2
```

Comment:
I don't intend to maintain python 2 compatibility, so I would instead remove this entirely.